packages/elements/src/slider/index.ts

```diff
@@ -575,29 +581,97 @@ export class Slider extends ControlElement {
   }
 
   /**
-   * Calculate percentage from value, min and max
-   * @param value value to be calculated
-   * @returns percentage
+   * Handles key down event on thumbs
+   * @param event Keyboard event
+   * @returns {void}
    */
-  private calculatePercentage (value: number): number {
-    const percentage = Math.abs((((value || 0) - this.minNumber) / (this.maxNumber - this.minNumber)) * 100);
-    if (percentage > 100) {
-      return 100;
+  private onKeyDown (event: KeyboardEvent): void {
+    if (this.readonly || this.minNumber >= this.maxNumber) {
+      return;
     }
-    else if (percentage < 0) {
-      return 0;
+
+    switch (event.key) {
+      case 'ArrowDown':
+      case 'Down':
+      case 'ArrowLeft':
+      case 'Left':
+        this.onApplyStep(Direction.Down);
+        break;
+      case 'ArrowUp':
+      case 'Up':
+      case 'ArrowRight':
+      case 'Right':
+        this.onApplyStep(Direction.Up);
+        break;
+      default:
+        return;
     }
-    else {
-      return percentage;
+
+    event.preventDefault();
+  }
+
+  /**
+   * Increase or decrease value depending on direction
+   * Then fires value change event
+   * @param direction Up or Down
+   * @returns {void}
+   */
+  private onApplyStep (direction: Direction): void {
+    // Get current thumb position and step in percentage format
+    const thumbPosition = this.calculatePercentage(this.valueNumber, 1);
```

Comment:
I think it makes more sense to call this function as `calculatePosition`. At least my assumption of calculating percents is `value / total * 100`, but the function clearly does more than that :)

Reply:
Thanks. I like the name. Updated.